ui/components/app/hold-to-reveal-button/hold-to-reveal-button.js

```diff
@@ -0,0 +1,180 @@
+import React, { useCallback, useContext, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../ui/button';
+import { I18nContext } from '../../../contexts/i18n';
+import Box from '../../ui/box/box';
+import {
+  ALIGN_ITEMS,
+  DISPLAY,
+  JUSTIFY_CONTENT,
+} from '../../../helpers/constants/design-system';
+
+const radius = 14;
+const strokeWidth = 2;
+const radiusWithStroke = radius - strokeWidth / 2;
+
+export default function HoldToRevealButton({ buttonText, onLongPressed }) {
+  const t = useContext(I18nContext);
+  const isLongPressing = useRef(false);
+  const [isUnlocking, setIsUnlocking] = useState(false);
+  const [hasTriggeredUnlock, setHasTriggeredUnlock] = useState(false);
+
+  /**
+   * Prevent animation events from propogating up
+   *
+   * @param e - Native animation event - React.AnimationEvent<HTMLDivElement>
+   */
+  const preventPropogation = (e) => {
+    e.stopPropagation();
+  };
+
+  /**
+   * Event for mouse click down
+   */
+  const onMouseDown = () => {
+    isLongPressing.current = true;
```

Comment:
Do you mind explaining how this works? All I'm seeing is that if you mouse down, this line gets run but I'm a bit confused as to how the other functions here end up getting called.

Reply:
You can check the ticket description (https://github.com/MetaMask/metamask-extension/issues/13260) and the conversation on Slack (https://consensys.slack.com/archives/C02GH833686/p1645710947856339) in order to get familiar with the specifics.
In short `React hooks and refs` are being used to detect and notify when a mouse is up or down. Depending on their states the component gets updated - calling the other functions, new CSS classes etc.

Reply:
Okay, gotcha. It looks like CSS animations power the callbacks that are taking place in the component. That's the piece I was missing :)